modules/ServerAPI/src/server/routes/ThingServiceRoutes.coffee

```diff
@@ -285,6 +287,42 @@ updateThing = (thing, testMode, callback) ->
 		)
 
 
+exports.bulkPostThingsSaveFile = (req, resp) ->
+	exports.bulkPostThingsInternal req.body, (response) =>
+		checkFilesAndUpdate = (thing, callback) ->
+			fileVals = serverUtilityFunctions.getFileValuesFromEntity thing, false
+			filesToSave = fileVals.length
+
+			completeThingUpdate = (thingToUpdate)->
+				updateThing thingToUpdate, false, (updatedThing) ->
+					callback updatedThing, 200
+
+			fileSaveCompleted = (passed) ->
+				if !passed
+					callback "file move failed", 500
+				if --filesToSave == 0 then completeThingUpdate(thing)
+
+			if filesToSave > 0
+				prefix = serverUtilityFunctions.getPrefixFromEntityCode thing.codeName
+				for fv in fileVals
+					csUtilities.relocateEntityFile fv, prefix, thing.codeName, fileSaveCompleted
+			else
+				callback thing, 200
+
+		if response.indexOf("saveFailed") > -1
```

Comment:
oh, this is where the actual function calls for the callback from `bulkPostThingsInternal` start.